draft-pda-protocol.md

```diff
@@ -573,116 +612,288 @@ PA protocol. The structure of the leader share is similarly protocol specific:
-The remainder of the me structure is dedicated to the protocol-specific helper
-share and request parameters used for the current round.
+(This field is marked optional because it is usually only needed for the first
+aggregate request. [TODO: Decide how to mark the helper input share as optional
+as well. This too is usually only needed for the first request.]) The `time`
+field should match the timestamp of the corresponding report. The remainder of
+the structure is dedicated to the protocol-specific helper share and request
+parameters used for the current round.
 
 The helper handles well-formed requests as follows. (As usual, malformed
 requests are handled as described in {{pa-error-common-aborts}}.) It first looks
-for the PA parameters `PAParam` for which `PAVerifyReq.task.id ==
+for the PA parameters `PAParam` for which `PAAggregateReq.task.id ==
 PAParam.task.id`. Next, it looks up the HPKE config and corresponding secret key
-associated with `PAVerifyReq.key_config_id`. If not found, then it aborts and
-alerts the leader with "unrecognized key config". [NOTE: In this situation, the
-leader has no choice but to abort. This falls into the class of error scenarios
-that are addressable by running with multiple helpers.]
+associated with `PAAggregateReq.helper_hpke_config_id`. If not found, then it
+aborts and alerts the leader with "unrecognized key config". [NOTE: In this
+situation, the leader has no choice but to abort. This falls into the class of
+error scenarios that are addressable by running with multiple helpers.]
 
-The response is structured as a sequence of *sub-responses*, where the i-th
-sub-response corresponds to the sub-request for each i. The structure of each
-sub-response is specific to the PA protocol:
+The response consists of the helper's updated state and a sequence of
+*sub-responses*, where the i-th sub-response corresponds to the sub-request for
+each i. The structure of each sub-response is specific to the PA protocol:
 
 ~~~
 struct {
-   PAVerifyResp seq<1..2^24-1>;
-} PAVerifyRespSeq;
+  opaque helper_state<0..2^16>;
+  PAAggregateSubResp seq<1..2^24-1>;
+} PAAggregateResp;
 
 struct {
-  PAProto;
-  select (PAVerifyResp.proto) {
-    case prio: PrioVerifyResp;
-    case hits: HitsVerifyResp;
+  PAProto proto;
+  select (PAAggregateSubResp.proto) {
+    case prio: PrioAggregateSubResp;
+    case hits: HitsAggregateSubResp;
   }
-} PAVerifyResp;
+} PAAggregateSubResp;
 ~~~
 
-For each sub-request `PAVerifyReq`, the helper computes the corresponding
-sub-response as follows. It first checks that checks that `PAVerifyReq.proto ==
-PAParam.proto`. If not, it aborts and alerts the leader with "incorrect protocol
-for sub-request". Otherwise, It computes the HPKE context as
+The helper handles each sub-request `PAAggregateSubReq` as follows.  It first
+checks that checks that `PAAggregateSubReq.proto == PAParam.proto`. If not, it
+aborts and alerts the leader with "incorrect protocol for sub-request".
```

Comment:
This could be avoided by just not having these fields.

Reply:
Ack, added a TODO.